### PR TITLE
Adjusts PDF export to be compatible with Vite and worker modules

### PR DESCRIPTION
**Pull request by mgcm**
The font initialization had the be reworked so it is compatible with how Vite and worker modules work in the standalone project.

**:heavy_check_mark: Checklist**

- [X] A changeset describing the change and affected packages ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#changelog-and-versioning)).
- [ ] Added or updated documentation.
- [ ] Tests for new functionality and regression tests for bug fixes.
- [ ] Screenshots or videos attached (for UI changes).
- [X] All your commits have a `Signed-off-by` line in the message ([more info](https://github.com/nordeck/.github/blob/main/docs/CONTRIBUTING.md#dco)).

### CODE REVIEW

File: packages/react-sdk/src/components/BoardBar/ExportWhiteboardDialogDownloadPdf.tsx [hunk 1]
```diff
@@ -97,9 +97,6 @@ function useGeneratePdf(
   useEffect(() => {
     let url: string | undefined = undefined;
 
-    setDownloadUrl(undefined);
-    onError?.(undefined);
```
Comment (mgcm) [inline packages/react-sdk/src/components/BoardBar/ExportWhiteboardDialogDownloadPdf.tsx:101]: This was causing a loop on export in Vite production builds.

File: packages/react-sdk/src/components/BoardBar/ExportWhiteboardDialogDownloadPdf.tsx [hunk 2]
```diff
@@ -114,14 +113,13 @@ function useGeneratePdf(
...
+        }
         const logger = getLogger('ExportWhiteboardDialogDownloadPdf');
         logger.error('Error while generating the PDF', e);
-        onError?.(
-          t(
-            'boardBar.exportWhiteboardDialog.pdfError',
-            'Something went wrong while generating the PDF.',
-          ),
-        );
+        onError?.(e.message);
```
Comment (mgcm) [inline packages/react-sdk/src/components/BoardBar/ExportWhiteboardDialogDownloadPdf.tsx:122]: moving this around as the error is generic and `t` was causing a re-render loop in production builds when used vite in the standalone project.

File: packages/react-sdk/src/components/BoardBar/ExportWhiteboardDialogDownloadPdf.tsx (hunk 1 above)
Comment (mgcm) [reply packages/react-sdk/src/components/BoardBar/ExportWhiteboardDialogDownloadPdf.tsx]: actually, no - see other comment.

File: packages/react-sdk/src/components/BoardBar/pdf/pdf.local.ts [hunk 3]
```diff
@@ -14,18 +14,17 @@
...
+import { initializeFonts } from './initializeFonts';
 
 /**
  * Generate the PDF in a separate module that can either:
  *   1. be imported by a web worker
  *   2. be imported lazily as a separate module
  */
 export function generatePdf(content: TDocumentDefinitions): Promise<Blob> {
-  const pdf = pdfMake.createPdf(content);
+  const cfg = initializeFonts();
```
Comment (weeman1337) [inline packages/react-sdk/src/components/BoardBar/pdf/pdf.local.ts:27]: I wonder if the name still make sense.
What about to rename it to `createFontConfig` ?

Comment (mgcm) [reply packages/react-sdk/src/components/BoardBar/pdf/pdf.local.ts]: Sure, makes sense.

**Review by weeman1337 — APPROVED** [review]

**Pull request merged**